tensornetwork/linalg/linalg.py

```diff
@@ -0,0 +1,191 @@
+from typing import Text
+import numpy as np
+from tensornetwork.backends import base_backend
+from tensornetwork import backend_contextmanager
+from tensornetwork import backends
+from tensornetwork import network_components
+
+
+Tensor = Any
+BaseBackend = base_backend.BaseBackend
+
+
+# INITIALIZATION
+def initialize_node(fname: Text,
+                    *fargs: Any,
+                    name: Optional[Text] = None,
+                    axis_names: Optional[List[Text]] = None,
+                    backend: Optional[Union[Text, BaseBackend]] = None,
+                    **fkwargs: Any):
+  """Return a Node wrapping data obtained by an initialization function
+  implemented in a backend. The Node will have the same shape as the
+  underlying array that function generates, with all Edges dangling.
+
+  This function is not intended to be called directly, but doing so should
+  be safe enough.
+  Args:
+    fname:  Name of the method of backend to call (a string).
+    *fargs: Positional arguments to the  initialization method.
+    name: Optional name of the Node.
+    axis_names: Optional names of the Node's dangling edges.
+    backend: The backend or its name.
+    **fkwargs: Keyword arguments to the initialization method.
+
+  Returns:
+    node: A Node wrapping data generated by
+          (the_backend).fname(*fargs, **kwargs), with one dangling edge per
+          axis of data.
+  """
+  if backend is None:
+    backend_obj = backend_contextmanager.get_default_backend()
+  else:
+    backend_obj = backends.backend_factory.get_backend(backend)
+  func = getattr(backend_obj, fname)
+  data = func(*fargs, **fkwargs)
+  node = network_components.Node(data, name=name, axis_names=axis_names,
+                                 backend=backend)
+  return node
+
+
+def eye(N: int,
+        dtype: Optional[Type[np.number]],
+        M: Optional[int] = None,
+        name: Optional[Text] = None,
+        axis_names: Optional[List[Text]] = None,
+        backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node representing a 2D array with ones on the diagonal and
+  zeros elsewhere. The Node has two dangling Edges.
+  Args:
+    N (int): The first dimension of the returned matrix.
+    dtype, optional: The dtype of the returned matrix.
+    M (int, optional): The second dimension of the returned matrix.
+    name (text, optional): Name of the Node.
+    axis_names (optional): List of names of the edges.
+    backend (optional): The backend or its name.
+
+  Returns:
+    I : Node of shape (N, M)
+        Represents an array of all zeros except for the k'th diagonal of all
+        ones.
+  """
+  the_node = initialize_node("eye", N,
+                             name=name, axis_names=axis_names, backend=backend,
+                             dtype=dtype, M=M)
+  return the_node
+
+
+def zeros(shape: Sequence[int],
+          dtype: Optional[Type[np.number]],
```

Comment:
should this have a default value?

Reply:
That seems reasonable, yes. What should it be?

Reply:
I would say double precision floats

Reply:
The default is now None, under the (tested) assumption that all backends provide a default value in this case.

Reply:
why not np.float64?

Reply:
then it would be clearer for the user

Reply:
Will this be valid for all backends (notably pytorch)?

Reply:
It also strikes me that the "best default" ought to be located at the backend; one could imagine that different backends might prefer different dtypes.

Reply:
I see, passing numpy dtypes using the backend initialization is problematic, yes. The default value though should be set within the top level function, and the user should be able to know what it is without looking into the backend documentation. Otherwise this is could be a source of really obscure bugs.

Reply:
Well, I guess I could just have a big if statement or something. For that to work, Pytorch would need to be imported at the top level, which would make it a dependency of the interface. Is that something we want?

Reply:
No we don't want that. My suggestion is to use numpy dtypes and convert them to the backend dtypes within the backend. Maybe I'm too picky, I just think that hidden choices about dtypes are really annoying.

Reply:
I'm fine with that, but then wouldn't the user need to consult the backend to figure out that the conversion had occurred?

Reply:
The idea is to use numpy to configure the dtypes (i.e. half vs single vs double), and convert this to the corresponding backend dtypes, (ie.e. np.floa64 will be converted to jax.numpy.float64, or torch.float64)

Reply:
We could give each backend a member "backend.default_dtype". Then, it would be obvious that each backend might do something different (rather than users perhaps assuming that their Nodes literally store np.float64), and while one would still need to look up the backend documentation to figure out what the default was, at least there would be only one clear place to look.

Reply:
@Thenerdstation

Reply:
> The idea is to use numpy to configure the dtypes (i.e. half vs single vs double), and convert this to the corresponding backend dtypes, (ie.e. np.floa64 will be converted to jax.numpy.float64, or torch.float64)

This also seems like a reasonable solution, provided it be documented appropriately.

Reply:
Jax, NumPy, and TensorFlow already behave in this way, so it seems only Pytorch would need to be modified.

Reply:
My opinion is to leave it as `None` and just stick to the default of the given backend, as that default option will usually be the most preformant choice for that backend.

I doubt there would be many "obsure bugs" related to a bad dtype. Infact, its actually more likely that the obsure bug would be a slow down do to defaulting to float64 instead of float32 for pytorch since float32 runs significantly faster on GPU hardware.

My preference is to leave it as `None` instead of adding a default dtype to the backends that the user can't control.

Reply:
In this case, pending Martin's agreement, all issues with this PR appear to be resolved

Reply:
I consider myself to be overuled :)